doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1162 @@
+- Include Tunnel counters as part of FlexCounter::collectCounters call. 
+
+
+## 5 CLI
+
+### 5.1 Click based CLI
+
+#### 5.1.1 Configuration Commands
+
+```
+1. VTEP Source IP configuration
+   - config vxlan add <vtepname> <src_ipv4>
+   - config vxlan del <vtepname>
+   - vtepname is a string. 
+   - src_ipv4 is an IPV4 address in dotted notation A.B.C.D
+2. EVPN NVO configuration 
+   - config vxlan evpn_nvo add <nvoname> <vtepname>
+   - config vxlan evpn_nvo del <nvoname>
+   - This configuration is specific to EVPN.
+3. VLAN VNI Mapping configuration
+   - config vxlan map add <vtepname> <vlanid> <vnid>
+   - config vxlan map_range add <vtepname>  <vlanstart> <vlanend> <vnistart>
+   - config vxlan map del <vtepname> <vlanid> <vnid>
+   - config vxlan map_range del <vtepname>  <vlanstart> <vlanend> <vnistart>
+4. VRF VNI Mapping configuration
+   - config vrf add_vrf_vni_map <vrf-name> <vni>
+   - config vrf del_vrf_vni_map  <vrf-name>
+5. ARP suppression
+   - config neigh-suppress vlan <vlan-id>  <"on"/"off">
+   - vlanid represents the vlan_id and on/off is enabled or disabled. By default ARP/ND suppression is disabled. 
+
+```
+
+#### 5.1.2 Show Commands
+
+```
+1. show vxlan interface 
+   - Displays the name, SIP, associated NVO name. 
+   - Displays the loopback interface configured with the VTEP SIP. 
+
+   VTEP Information:
+
+           VTEP Name : VTEP1, SIP  : 4.4.4.4
+           NVO Name  : nvo1,  VTEP : VTEP1
+           Source interface  : Loopback33
+
+2. show vxlan vlanvnimap 
+   - Displays all the VLAN VNI mappings.
+
+   +---------+-------+
+   | VLAN    |   VNI |
+   +=========+=======+
+   | Vlan100 |   100 |
+   +---------+-------+
+   | Vlan101 |   101 |
+   +---------+-------+
+   Total count : 2
+
+
+3. show vxlan vrfvnimap 
+   - Displays all the VRF VNI mappings.
+
+4. show vxlan tunnel
+   - lists all the discovered tunnels.  
+   - SIP, DIP, Creation Source, OperStatus are the columns.
+
+   +---------+---------+-------------------+--------------+
+   | SIP     | DIP     | Creation Source   | OperStatus   |
+   +=========+=========+===================+==============+
+   | 2.2.2.2 | 4.4.4.4 | EVPN              | oper_up      |
+   +---------+---------+-------------------+--------------+
+   | 2.2.2.2 | 3.3.3.3 | EVPN              | oper_up      |
+   +---------+---------+-------------------+--------------+
+   Total count : 2
+
+5. show vxlan remote_mac <remoteip/all>  <vlanid/all> 
+   - lists all the MACs learnt from the specified remote ip or all the remotes for the specified/all vlans. (APP DB view) 
+   - VLAN, MAC, RemoteVTEP,  VNI,  Type are the columns.
+
+   show vxlan remote_mac all
+   +---------+-------------------+--------------+-------+--------+
+   | VLAN    | MAC               | RemoteVTEP   |   VNI | Type   |
+   +=========+===================+==============+=======+========+
+   | Vlan101 | 00:00:00:00:00:01 | 4.4.4.4      |  1001 | static |
```

Comment:
Here the remote MAC type "static" conveys a meaning that, the learnt MAC is static on the Remote VTEP. This needs to be changed??